lib/ezutils/classes/ezextension.php

```diff
@@ -544,10 +544,10 @@ public static function getHandlerClass( ezpExtensionOptions $options )
             // we rely on the autoload system here
             if ( class_exists( $handler ) )
             {
-                // only use reflection if we have params to avoid exception on objects withouth constructor
-                if ( $handlerParams !== null && is_array( $handlerParams ) && count( $handlerParams ) > 0 )
+                $reflection = new ReflectionClass( $handler );
+                // only pass parameters if the class has a constructor
+                if ( $reflection->getConstructor() !== null && $handlerParams !== null && is_array( $handlerParams ) && count( $handlerParams ) > 0 )
                 {
-                    $reflection = new ReflectionClass( $handler );
```

Comment:
It could be notice or debug as well.
Point was to make sure developers are made aware of missing constructors by mistake, which before this PR would give you a clear `Uncaught PHP Exception `, but now it is silenced.